app/models/coupon.rb

```diff
@@ -0,0 +1,6 @@
+class Coupon < ApplicationRecord
+  belongs_to :user
```

Comment:
not sure about this; can we have the coupon belonging to one user, but also having many users? is that contradicting?